tests/edit/endpoint.test.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+var test = require('tape');
+
+var initServer = require('../../example/server.js');
+var config = require('../../config/load-config.js');
+var challenges = require('../../example/data/challenges.json');
+
+test('edit challenge function', function (t) {
+  var chalId = 2;
+  var updatedChal = {
+    title: 'A more specific Challenge Number 2',
+    description: 'I have looked into the possibilities of xyz and wonder whether...'
+  };
+  var req = {
+    method: 'POST',
+    url: '/edit?id=' + chalId,
+    payload: updatedChal
+  };
+
+  initServer(config, function (err, server, pool) {
+    if (err) {
+      return t.fail('Error starting the server, error: ', err);
+    }
+    return server.inject(req, function (res) {
+      t.deepEqual(res.payload, '[]', 'challenge has been editted!');
```

Comment:
No, we get that from getById - so the id is returned at the point where we get the existing challenge details (at GET endpoint) - because we POST back to the same url, we get the id from there.